### PR TITLE
On ASUS Zephyrus GA402X, make enabling auto-suspend on the keyboard optional

### PR DESCRIPTION
**Pull request by mexisme**
###### Description of changes

The bug that seemed to make the internal keyboard on the ASUS Zephyrus GA402X take 1-2 keypresses after suspend seems to be fixed in (at least) kernel 6.9.x.

This adds an option to select to turn the auto-suspend on or off for this device.
- On kernels before 6.9.x, default to disabling auto-suspend
- On more-recent kernels, default to enabling auto-suspend

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: asus/zephyrus/ga402x/shared.nix [hunk 1]
```diff
@@ -5,7 +5,11 @@
 }:
 
 let
-  inherit (lib) mkDefault mkIf mkMerge version versionOlder;
+  inherit (lib) mkDefault mkEnableOption mkIf mkMerge version versionAtLeast versionOlder;
+
+  cfg = config.hardware.asus.zephyrus.ga402x;
+  defaultAutosuspendEnable = if (versionAtLeast version "6.9") then true else false;
```
Comment (pbsds) [inline asus/zephyrus/ga402x/shared.nix:11]: you're testing the nixos version, not the kernel version
```suggestion
  defaultAutosuspendEnable = versionAtLeast config.boot.kernelPackages.kernel.version "6.9";
```

File: asus/zephyrus/ga402x/shared.nix [hunk 2]
```diff
@@ -43,15 +58,20 @@ in {
...
-            # Otherwise, it will tend to take 1-2 key-presses to wake-up after suspending
-            ACTION=="add", SUBSYSTEM=="usb", TEST=="power/autosuspend", ATTR{idVendor}=="0b05", ATTR{idProduct}=="19b6", ATTR{power/autosuspend}="-1"
-          '';
         };
       };
     }
 
+    (mkIf (! cfg.keyboard.autosuspend.enable) {
+      services.udev = {
+        extraRules = ''
```
Comment (pbsds) [inline asus/zephyrus/ga402x/shared.nix:67]: the `mkIf` could have been placed here in the prior version, causing a smaller diff overall
```suggestion
        extraRules = mkIf (! cfg.keyboard.autosuspend.enable) ''
```

Comment (mexisme) [reply asus/zephyrus/ga402x/shared.nix:67]: I'll try to rework it a little further to incorporate this.

Comment (mexisme) [reply asus/zephyrus/ga402x/shared.nix]: > the `mkIf` could have been placed here in the prior version, causing a smaller diff overall

Does this work better, @pbsds ?

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**